lxd/util/net.go

```diff
@@ -84,6 +84,9 @@ func CanonicalNetworkAddress(address string, defaultPort int) string {
 			// a port number, so append the default port.
 			address = fmt.Sprintf("%s:%d", address, defaultPort)
 		}
+	} else if port == "" && address[len(address)-1] == ':' {
+		// An address that ends with a trailing colon will be parsed as having an empty port.
+		address = net.JoinHostPort(host, fmt.Sprintf("%d", defaultPort))
```

Comment:
Is that going to cause us some issues when passed an IPv6 address like `::` `1234::`?

Reply:
Hmm, no, I guess it will be fine, the IPv6 case should parse as a valid address and so hit the above case.
The only time where we should hit this fallback logic is if there's a single trailing colon.